glean-core/src/core/mod.rs

```diff
@@ -269,6 +297,88 @@ impl Glean {
             ping_lifetime_max_time,
         )?);
 
+        {
+            // We just set it.
+            let data_store = glean.data_store.as_ref().unwrap();
+            let db_load_sizes = data_store.load_sizes.as_ref().unwrap();
+            let new_size = db_load_sizes.new.unwrap_or(0);
+
+            // If we have a client ID on disk, we check the database
+            if let Some(stored_client_id) = stored_client_id {
+                if new_size <= 0 {
+                    log::trace!("no database. database size={new_size}. stored_client_id={stored_client_id}");
+                    glean
+                        .health_metrics
+                        .recovered_client_id
+                        .set_from_uuid_sync(&glean, stored_client_id);
+                    glean
+                        .health_metrics
+                        .exception_state
+                        .set_sync(&glean, ExceptionState::EmptyDb);
+                } else {
+                    let db_client_id = glean
+                        .core_metrics
+                        .client_id
+                        .get_value(&glean, Some("glean_client_info"));
+
+                    match db_client_id {
+                        None => {
+                            log::trace!("no client_id in DB. stored_client_id={stored_client_id}");
+                            glean
+                                .health_metrics
+                                .exception_state
+                                .set_sync(&glean, ExceptionState::RegenDb);
+                        }
+                        Some(db_client_id) if db_client_id == *KNOWN_CLIENT_ID => {
```

Comment:
We've stopped storing `c0ffee` in db since, what, v64? When should we stop accounting for it?